src/all/hitomi/src/eu/kanade/tachiyomi/extension/all/hitomi/Hitomi.kt

```diff
@@ -657,11 +685,53 @@ class Hitomi(
         return hash.replace(Regex("""^.*(..)(.)$"""), "$2/$1")
     }
 
+    override fun setupPreferenceScreen(screen: PreferenceScreen) {
+        ListPreference(screen.context).apply {
+            key = PREF_IMAGETYPE
+            title = "Images Type"
+            entries = arrayOf("webp", "avif", "jxl")
+            entryValues = arrayOf("webp", "avif", "jxl")
+            summary = "Clear chapter cache to apply changes"
+            setDefaultValue("webp")
+        }.also(screen::addPreference)
+    }
+
+    private fun List<Int>.toBytesList(): ByteArray = this.map { it.toByte() }.toByteArray()
+    private val signatureOne = listOf(0xFF, 0x0A).toBytesList()
+    private val signatureTwo = listOf(0x00, 0x00, 0x00, 0x0C, 0x4A, 0x58, 0x4C, 0x20, 0x0D, 0x0A, 0x87, 0x0A).toBytesList()
+    fun ByteArray.startsWith(byteArray: ByteArray): Boolean {
+        if (this.size < byteArray.size) return false
+        return this.sliceArray(byteArray.indices).contentEquals(byteArray)
+    }
+
+    private fun Intercept(chain: Interceptor.Chain): Response {
+        val response = chain.proceed(chain.request())
+        if (response.headers["Content-Type"] != "application/octet-stream") {
+            return response
+        }
+
+        val bytesArray = response.peekBody(Long.MAX_VALUE).bytes()
```

Comment:
while this works you could optimize it to only peek the necessary bytes

Reply:
A possible implementation of this would be:
```suggestion
        val bytesPeek = max(signatureOne.size, signatureTwo.size).toLong()
        val bytesArray = response.peekBody(bytesPeek).bytes()
```